crates/polars-plan/src/logical_plan/optimizer/predicate_pushdown/mod.rs

```diff
@@ -191,12 +191,25 @@ impl<'a> PredicatePushDown<'a> {
 
         match lp {
             Selection { predicate, input } => {
+                // Correctness: It is maintained that acc_predicates does not have
+                // boundary predicates at this point.
+                insert_and_combine_predicate(&mut acc_predicates, predicate, expr_arena);
 
-                // If a predicates result would be influenced by earlier applied filter
-                // we remove it and apply it locally
-                let local_predicates = transfer_to_local_by_node(&mut acc_predicates, |node| predicate_is_pushdown_boundary(node, expr_arena));
+                // If a predicates result is influenced by predicates occurring
+                // before it, then it and all accumulated predicates must be
+                // blocked at this level.
+                let local_predicates = if acc_predicates
```

Comment:
Doesn't swapping resolve it. E.g. wouldn't keeping `transfer_to_local_by_node` here work?

As it is we are now removing all predicates even orthogonal ones.

Reply:
Nevermind, looking at this now.